### PR TITLE
feat:add page for non-EVM implementation reference

### PR DESCRIPTION
**Pull request by aroralanuk**

### CODE REVIEW

File: docs/protocol/non-evm-implementations.mdx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+# CosmWasm
+
+See the CosmWasm implementation of Hyperlane [here](https://github.com/many-things/cw-hyperlane), and follow [this guide](https://github.com/many-things/cw-hyperlane/blob/main/DEPLOYMENT.md) on how to deploy to a Cosmos appchain here. We're live on Celestia, Stride and Neutron as of July 2024.
+
+# Sealevel
+
+Hyperlane supports Solana Virtual Machine (SVM), and is currently live supporting Zebec's [Nautilus Bridge](https://medium.com/hyperlane/case-study-interoperability-for-the-modular-rollup-stack-featuring-nautilus-d01118fe31ec). In addition, it is live on [Eclipse testnet](https://medium.com/hyperlane/hyperlane-expands-to-solana-vm-in-collaboration-with-eclipse-8465616fd0e4).
+
+You can find the [Sealevel contract implementation here](https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/main/rust/sealevel/programs/mailbox/src/processor.rs).
```
Comment (tkporter) [inline docs/protocol/non-evm-implementations.mdx:11]: while we're at it imo it's fine to just point to `https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/main/rust/sealevel/programs`

File: docs/protocol/non-evm-implementations.mdx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+# Cairo
+
+Hyperlane protocol has been implemented in Cairo for Starkware by the [Pragma team](https://www.pragma.build/). You can find the [implementation here](https://github.com/astraly-labs/hyperlane_starknet).
```
Comment (tkporter) [inline docs/protocol/non-evm-implementations.mdx:15]: nit
```suggestion
The Hyperlane protocol has been implemented in Cairo for Starkware by the [Pragma team](https://www.pragma.build/). You can find the [implementation here](https://github.com/astraly-labs/hyperlane_starknet).
```

File: docs/protocol/non-evm-implementations.mdx [hunk 3]
```diff
@@ -0,0 +1,23 @@
+While our public documentation currently focuses on EVM compatibility, we offer varying levels of support for additional ecosystems. These ecosystems may not have full feature parity or complete tooling integration (such as TypeScript CLI support) at this time. We're actively working to expand our documentation and tooling across these platforms. If you're excited about building on top of these or bringing Hyperlane to other new ecosystems reach out on [our Discord](https://discord.gg/hyperlane)!
+
+# CosmWasm
+
+See the CosmWasm implementation of Hyperlane [here](https://github.com/many-things/cw-hyperlane), and follow [this guide](https://github.com/many-things/cw-hyperlane/blob/main/DEPLOYMENT.md) on how to deploy to a Cosmos appchain here. We're live on Celestia, Stride and Neutron as of July 2024.
```
Comment (tkporter) [inline docs/protocol/non-evm-implementations.mdx:5]: wonder if we should give explicit credit to Mitosis as we do for the other implementations?

File: docs/protocol/non-evm-implementations.mdx [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+Hyperlane protocol has been implemented in Cairo for Starkware by the [Pragma team](https://www.pragma.build/). You can find the [implementation here](https://github.com/astraly-labs/hyperlane_starknet).
+
+# Move
+
+Hyperlane has been implemented in MoveVM by [Movement Labs](https://movementlabs.xyz/). You can find the [implementation here](https://github.com/movementlabsxyz/hyperlane-monorepo). These Move contracts are live on Movement's M1 testnet.
+
+# Sway
+
+Find our [Sway implementation here](https://github.com/hyperlane-xyz/fuel-contracts/blob/main/contracts/hyperlane-mailbox/src/main.sw).
```
Comment (tkporter) [inline docs/protocol/non-evm-implementations.mdx:23]: Imo pointing to https://github.com/hyperlane-xyz/fuel-contracts is fine

File: docs/guides/implementation-guide.mdx [hunk 5]
```diff
@@ -267,19 +273,21 @@ function getAnnouncedStorageLocations(
...
 Implements a security module that checks if the metadata provided to verify satisfies a quorum of signatures from a set of configured validators.
 
-* [Solidity](https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/main/solidity/contracts/isms/multisig/AbstractMultisigIsm.sol)
-* [Sway](https://github.com/hyperlane-xyz/fuel-contracts/blob/main/contracts/multisig-ism/src/main.sw)
-* [CosmWasm](https://github.com/hyperlane-xyz/cosmwasm/blob/main/packages/interface/src/ism/multisig.rs)
-* [Sealevel](https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/main/rust/sealevel/libraries/multisig-ism/src/interface.rs)
+- [Solidity](https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/main/solidity/contracts/isms/multisig/AbstractMultisigIsm.sol)
+- [Sway](https://github.com/hyperlane-xyz/fuel-contracts/blob/main/contracts/multisig-ism/src/main.sw)
+- [CosmWasm](https://github.com/many-things/cw-hyperlane/blob/main/packages/interface/src/ism/multisig.rs)
+- [Sealevel](https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/main/rust/sealevel/libraries/multisig-ism/src/interface.rs)
```
Comment (tkporter) [inline docs/guides/implementation-guide.mdx:279]:
```suggestion
- [Sealevel](https://github.com/hyperlane-xyz/hyperlane-monorepo/tree/main/rust/sealevel/programs/ism/multisig-ism-message-id)
```
small nit, this is the actual program and the other one is a library that the program uses

**Review by tkporter — APPROVED** [review]

File: docs/protocol/non-evm-implementations.mdx (hunk 3 above)
Comment (aroralanuk) [reply docs/protocol/non-evm-implementations.mdx:5]: I think we should

**Pull request merged**